### ISSUE TITLE
tf.keras.experimental.WideDeepModel  can not accept two  tf.keras.Model  as input?

### ISSUE THREAD

**Issue by jiaqinglin** [tensorflow/tensorflow#44640]
tf.keras.experimental.WideDeepModel claims:

`tf.keras.experimental.WideDeepModel(linear_model, dnn_model, activation=None, **kwargs)`
Args:
linear_mode | a premade LinearModel, its output must match the output of the dnn model.
dnn_model   | a tf.keras.Model, its output must match the output of the linear model.
activation     | Activation function. Set it to None to maintain a linear activation.
**kwargs      | The keyword arguments that are passed on to BaseLayer.init. Allowed keyword arguments include name.

I viewd the source code and can not find the difference between linear model and deep model.
When i used this WideDeepModel with two ' tf.keras.Model' , i got some problem.

```
# input_dict key is the feature name,  value is keras Input Layer
deep_embedding_output=layers_utils.get_embed_output(input_dict, embedding_size=16)
deep_embedding_flat_inp = layers.Flatten(deep_embedding_output)
deep_mlp_out= layers_utils.mlp(units=[128,64,16])(deep_embedding_flat_inp)
deep_out=layers.Dense(1)(deep_mlp_out)
deep_channel = Model(inputs=input_dict,output=[deep_out])

wide_embedding_output=layers_utils.get_embed_output(input_dict, embedding_size=1)
wide_embedding_flat_inp = layers.Flatten(wide_embedding_output)
wide_out=layers.Dense(1)(wide_embedding_flat_inp)
deep_channel = Model(inputs=input_dict,output=[wide_out])
# label feature name is label
model = WideDeepModel(wide_channel, deep_channel, act="sigmoid", name=‘label’)

model.compile(optimizer=['ftrl', 'adam'], 'binary_cross_entropy', ['auc'])
# `tf.data.Dataset` that contains a single tensor
combined_model.fit(dataset, epochs)
#dataset = tf.data.Dataset.from_tensors((features, label))
```
I met the problem:
linear model expected 94 ,but given 95;
94 is the input feature size, it seems like that the target label was not recognized?

**Comment by jiaqinglin** [comment]
when i used this tf.data.Dataset as follows:
```
deep_embedding_output=layers_utils.get_embed_output(input_dict, embedding_size=16)
deep_embedding_flat_inp = layers.Flatten(deep_embedding_output)
deep_mlp_out= layers_utils.mlp(units=[128,64,16])(deep_embedding_flat_inp)
deep_out=layers.Dense(1)(deep_mlp_out)

wide_embedding_output=layers_utils.get_embed_output(input_dict, embedding_size=1)
wide_embedding_flat_inp = layers.Flatten(wide_embedding_output)
wide_out=layers.Dense(1)(wide_embedding_flat_inp)

wd =  layers.Add()([wide_out,deep_out])
wd_out = layers.Dense(1,act='sigmoid',name='label')(wd)
model =  Model(inputs=input_dict,output=[wd_out])
model.compile(optimizer='adam', 'binary_cross_entropy', ['auc'])
combined_model.fit(dataset, epochs)
```
This model can work.

If i tried to remove `wd_out = layers.Dense(1,act='sigmoid',name='label')(wd)` the name argument,  it did not work.
So i guss that my problem may caused by 'label' tag.

**Comment by jiaqinglin** [comment]
@tanzhenyu Thx

**Comment by tanzhenyu** [comment]
It sounds like this is a user code issue and we're safe to close it?

**Comment by ravikyram** [comment]
@jiaqinglin 

Is this still an issue?Please, close this thread if your issue was resolved. Thanks!
